### DISCUSSION TITLE
[Bug]: can not copy code block (edge case)

### DISCUSSION THREAD

**Discussion by Originalimoc** [danny-avila/LibreChat#675] [Troubleshooting]
### What happened?

When you click copy code, this happens:
```
CodeBlock.tsx:25 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'writeText')
    at onClick (CodeBlock.tsx:25:35)
    at Object.mD (vendor-e5aad502.js:37:9864)
    at vD (vendor-e5aad502.js:37:10018)
    at ED (vendor-e5aad502.js:37:10075)
    at Y8 (vendor-e5aad502.js:37:31482)
    at p_ (vendor-e5aad502.js:37:31899)
    at vendor-e5aad502.js:37:36812
    at qE (vendor-e5aad502.js:40:36956)
    at QS (vendor-e5aad502.js:37:8999)
    at f4 (vendor-e5aad502.js:37:33186)
```

### Steps to Reproduce

1. Ask LLM to generate some code.
2. Click copy

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Originalimoc** [comment]
Well after move it behind a 443 port HTTPS reverse proxy it works...? Plain http://IP:3080 doesn't work.

**Comment by danny-avila** [comment]
This is a weird edge case. I use it without HTTPS and I'm able to copy it just fine.

![image](https://github.com/danny-avila/LibreChat/assets/110412045/4d8d7580-31b4-4d92-88ca-3140cd047c85)

My guess is that something is happening while you are building the frontend, how are you running the app?

**Comment by Originalimoc** [comment]
Docker. On same session: Direct HTTP: this issue and login does not persist a page reload. Behind Apache HTTPS: both issue not showing. It's http://172.xx.xxLAN:3080

**Comment by danny-avila** [comment]
I’m afraid I can’t reproduce your issues. Would you please try a clean installation with docker, with default domain/port configuration and let me know how that works? Perhaps you can then incrementally change your settings and see which produces the error

**Reply by danny-avila** [reply]
I recommend you update your app to latest main branch.

⚠️ This assumes you made no changes to the main branch and you have at least npm and git installed.
```bash
git fetch origin
git checkout main
git pull origin main
```

Then to update, you have 3 options: 
```bash
npm run update #will prompt user if updating docker or local installation
# OR
npm run update:local
npm run update:docker
```
